commons/src/main/scala/com/lynbrookrobotics/potassium/commons/flywheel/DoubleFlywheel.scala

```diff
@@ -0,0 +1,91 @@
+package com.lynbrookrobotics.potassium.commons.flywheel
+
+import com.lynbrookrobotics.potassium.{Component, PeriodicSignal, Signal, SignalLike}
+import com.lynbrookrobotics.potassium.control.{PIDF, PIDFConfig}
+import com.lynbrookrobotics.potassium.tasks.{ContinuousTask, FiniteTask}
+import com.lynbrookrobotics.potassium.units.{GenericDerivative, GenericValue}
+
+import squants.{Angle, Dimensionless}
+import squants.motion.AngularVelocity
+
+trait DoubleFlywheelProperties {
+  def velocityGains: PIDFConfig[AngularVelocity,
+                                GenericValue[AngularVelocity],
+                                AngularVelocity,
+                                GenericDerivative[AngularVelocity],
+                                Angle,
+                                Dimensionless]
+}
+
+trait DoubleFlywheelHardware {
+  def leftVelocity: Signal[AngularVelocity]
+  def rightVelocity: Signal[AngularVelocity]
+}
+
+abstract class DoubleFlywheel {
+  type Properties <: DoubleFlywheelProperties
+  type Hardware <: DoubleFlywheelHardware
+
+  case class DoubleFlywheelSignal(left: Dimensionless, right: Dimensionless)
+
+  def outputSignal(s: DoubleFlywheelSignal)(implicit hardware: Hardware): Unit
+
+  object velocityControllers {
+    def velocityControl(target: AngularVelocity)
+                       (implicit properties: Signal[Properties],
+                        hardware: Hardware): (Signal[AngularVelocity], Signal[AngularVelocity], PeriodicSignal[DoubleFlywheelSignal]) = {
+      val errorLeft = hardware.leftVelocity.map(target - _)
+      val errorRight = hardware.rightVelocity.map(target - _)
+
+      val controlLeft = PIDF.pidf(
+        hardware.leftVelocity.toPeriodic,
+        Signal.constant(target).toPeriodic,
+        properties.map(_.velocityGains)
+      )
+
+      val controlRight = PIDF.pidf(
```

Comment:
One controls with the left velocity, the other with the right, so we want to keep them separate.